File: Java/refactoring_exercise_3/src/main/java/za/co/entelect/refactoring3/domain/SavingsAccount.java
package za.co.entelect.refactoring3.domain;

public class SavingsAccount extends BankAccount {

    public SavingsAccount(Long balanceInCents, double creditInterestsRate, double debitInterestRate, long accountFee) {
        super(balanceInCents, creditInterestsRate, debitInterestRate, accountFee);
    }

    @Override
    public AccountType getAccountType() {
        return AccountType.SAVINGS;
    }
}

File: C#/Refactoring/refactoring_exercise_3/za/co/entelect/refactoring3/domain/SavingsAccount.cs
namespace refactoring_exercise_3.za.co.entelect.refactoring3.domain
{
    public class SavingsAccount : BankAccount
    {

        public SavingsAccount(long balanceInCents, double creditInterestsRate, double debitInterestRate, long accountFee) :
            base(balanceInCents, creditInterestsRate, debitInterestRate, accountFee)
        {

        }

        public override AccountType GetAccountType()
        {
            return AccountType.Savings;
        }

    }
}